src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -0,0 +1,231 @@
+                this.addAcl(searchValue)
+              }
+            },
+            suggestions: _.difference(suggestions, _.map('email', acl)),
+            style: { fontSize: 16, padding: '0 1rem 0 0', height: 'unset', border: 'none' },
+            focus: { border: 'none' },
+            theme: { suggestion: { padding: 0 } }
+          }
+        })
+      ]),
+      div({ style: styles.currentCollaboratorsArea }, [
+        div({ style: Style.elements.sectionHeader }, ['Current Collaborators']),
+        ...acl.map(this.renderCollaborator),
+        !loaded && centeredSpinner()
+      ]),
+      updateError && div({ style: { marginTop: '1rem' } }, [
+        div({}, ['An error occurred:']),
+        updateError
+      ]),
+      working && spinnerOverlay
+    ])
+  }
+
+  addAcl(email) {
+    const { acl } = this.state
+    this.setState({ acl: _.concat(acl, [{ email, accessLevel: 'READER', pending: false }]), searchValue: '' })
+  }
+
+  renderCollaborator = ({ email, accessLevel, pending }, index) => {
+    const isPO = accessLevel === 'PROJECT_OWNER'
+    const isMe = email === getBasicProfile().getEmail()
+    const { acl } = this.state
+
+    return div({
+      style: { display: 'flex', padding: '0.5rem', borderTop: index && `1px solid ${colors.gray[4]}` }
+    }, [
+      div({
+        style: { flex: 1 }
+      }, [
+        div({}, [email]),
+        pending && div({ style: styles.pending }, ['Pending']),
+        isPO ?
+          h(Select, {
+            wrapperStyle: styles.roleSelect,
+            disabled: true,
+            value: 'Project Owner',
+            options: [{ label: 'Project Owner', value: 'Project Owner' }]
+          }) :
+          h(Select, {
+            wrapperStyle: styles.roleSelect,
+            searchable: false, clearable: false,
+            disabled: isMe,
+            value: accessLevel,
+            onChange: ({ value }) => this.setState({ acl: _.set([index, 'accessLevel'], value, acl) }),
+            options: _.map(level => ({ label: level, value: level.toUpperCase() }), ['Reader', 'Writer', 'Owner'])
+          })
+      ]),
+      !isPO && !isMe && linkButton({
+        onClick: () => this.setState({ acl: _.remove({ email }, acl) })
+      }, [icon('minus-circle', { size: 24 })])
+    ])
+  }
+
+  async componentDidMount() {
+    const { namespace, name, onDismiss } = this.props
+
+    try {
+      const [{ acl }, shareSuggestions, groups] = await Promise.all([
+        Workspaces.workspace(namespace, name).getAcl(),
+        Workspaces.getShareLog(),
+        Groups.list()
+      ])
+
+      const fixedAcl = _.flow(
+        _.toPairs,
+        _.map(([email, { pending, accessLevel }]) => ({ email, pending, accessLevel })),
+        _.sortBy(x => -Utils.workspaceAccessLevels.indexOf(x.accessLevel))
+      )(acl)
+
+      this.setState({
+        acl: fixedAcl,
+        originalAcl: fixedAcl,
+        groups,
+        shareSuggestions,
+        loaded: true
+      })
```

Comment:
Nice state management. Keeping the original and modified data makes it straightforward to handle edits, and then compute the diff at the end.